tests/unit-tests/src/test/java/org/apache/activemq/artemis/tests/unit/core/postoffice/impl/AddressMapUnitTest.java

```diff
@@ -638,7 +638,7 @@ public void testHashA() throws Exception {
       assertFalse(aABCA.matches(aHashA));
       assertFalse(aHashA.matches(aABCA));
 
-      assertEquals(0, countMatchingWildcards(abcaS));
+      assertEquals(1, countMatchingWildcards(abcaS));
```

Comment:
This doesnt seem right. I would expect existing test case to pass the same without modification

Reply:
Isn't this test essentially about if "#.a" matches "a.b.c.a"? Which I think it should.

Reply:
My issue is that would be a change of behaviour, that some users may have got used to and built systems around. As such just changing that may break many users.

If theres a change in behaviour it needs to be behind a toggle and then by default kept as original.

Reply:
@michaelandrepearce this is a bug fix for a regression, the tests verify the behaviour of the existing code but the semantic was wrong in the tests.